tests/queryHas.js

```diff
@@ -28,4 +28,11 @@ describe('Parent selector query', function () {
         assert.equal(0, matches.length);
     });
 
+    it('binary op', function () {
+        const deepChildMatches = esquery(conditional, 'IfStatement:has(> Identifier[name="x"])');
+        assert.equal(0, deepChildMatches.length);
+
+        const shallowChildMatches = esquery(conditional, 'IfStatement:has(> LogicalExpression.test, > Identifier[name="x"])');
```

Comment:
hey I tried to use a more complex version of this:
```ts
    it('fails repeated and nested', function () {
        const matches = esquery(
            esprima.parse(`
                if (x) foo; // Target
                if (x) { foo; }
                if (y) { foo; }
                if (x) bar;
            `),
            'IfStatement:has(>Identifier[name="x"]):has(>ExpressionStatement>Identifier[name="foo"])'
        );
        assert.equal(1, matches.length);
    });
```
and it doesn't work. Any ideas why, in case this is still fresh on your mind? I found this as a workaround in the mean time:
```ts
    it('workaround', function () {
        const matches = esquery(
            esprima.parse(`
                if (x) foo; // Target
                if (x) { foo; }
                if (y) { foo; }
                if (x) bar;
            `),
            'IfStatement:has(>Identifier[name="x"]):has(>ExpressionStatement:has(>Identifier[name="foo"]))'
        );
        assert.equal(1, matches.length);
    });
```

and extracted a simpler to reproduce test case into https://github.com/estools/esquery/pull/146

Reply:
More serious bug:
`> LogicalExpression.test`
your test should find 0 matches. LogicalExpression never has a test field

Reply:
Hm sorry LogicalExpression.test found matches even before version 1.5

Reply:
An attempt to fix the complex queries https://github.com/estools/esquery/pull/148